src/main/host/descriptor/channel.c

```diff
@@ -133,6 +140,12 @@ static gssize channel_receiveUserData(Transport* transport, PluginVirtualPtr buf
     /* the write end of a unidirectional pipe can not read! */
     utility_assert(channel->type != CT_WRITEONLY);
 
+    /* Zero-size reads of pipes aren't very clearly specified, but
+     * experimentally they are a no-op. */
+    if (nBytes == 0) {
+        return 0;
+    }
```

Comment:
I'm not 100% sure if this is correct for reading. If this is a non-blocking pipe, it might first check if there are bytes to read and return EAGAIN if not.

Reply:
I added a second size-zero read to the test, which validates that it doesn't block. I think it's reasonable to infer it would also not return EAGAIN for a nonblocking descriptor.